### PR TITLE
Feature/#5 copy uploaded files

### PR DESCRIPTION
**Pull request by infeo**
This PR addresses #5 by implementing the "copy and forget" strategy described as the (preferred) solution in the issue.

### CODE REVIEW

File: src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java [hunk 1]
```diff
@@ -1,59 +1,85 @@
...
+	 */
 	public CompletionStage<Void> scheduleUpload(OpenFile file) {
 		if (!file.isDirty()) {
 			LOG.trace("Upload of {} skipped. Unmodified.", file.getPath());
 			return CompletableFuture.completedFuture(null);
 		}
 		try {
-			var in = file.asPersistableStream();
-			return scheduleUpload(file, in);
+			Path toUpload = cacheDir.resolve(file.getPath().getFileName().toString() + ".tmp");
```
Comment (overheadhunter) [inline src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java:56]: The name should be ~more~ unique. There can be multiple pending uploads for different `foo/example.txt` and `bar/example.txt`. A UUID should do the job.

File: src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java [hunk 2]
```diff
@@ -1,59 +1,85 @@
...
+			Files.copy(file.asPersistableStream(), toUpload, StandardCopyOption.REPLACE_EXISTING);
```
Comment (overheadhunter) [inline src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java:57]: No REPLACE_EXISTING needed, if we assert the tmp file to be unique

File: src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java [hunk 3]
```diff
@@ -1,59 +1,85 @@
...
+			file.setDirty(false);
+			return scheduleUpload(file.getPath(), Files.newInputStream(toUpload, StandardOpenOption.DELETE_ON_CLOSE));
```
Comment (overheadhunter) [inline src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java:59]: I like the idea of `DELETE_ON_CLOSE`, but I'm not sure this works as intended on all OS.

Comment (overheadhunter) [reply src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java:59]: Doesn't mean this should be changed, but needs to be tested ;-)

CC @tobihagemann

**Review by overheadhunter — CHANGES_REQUESTED** [review]
Just not sure regarding the name of the tmp files.

Also, since this change affects OpenFiles as well, shouldn't we add some tests to make sure, modifications to OpenFiles aren't impaired by uploads?

File: src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java (hunk 1 above)
Comment (infeo) [reply src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java:56]: If we use `UUID::fromString` we can use the (unique) cloud path as an (indirect) identifier. With this we would avoid having several, updated copies of the file and just replace the current one with the most recent change.

**Comment by infeo** [comment]
> Also, since this change affects OpenFiles as well, shouldn't we add some tests to make sure, modifications to OpenFiles aren't impaired by uploads?

Yes, that is recommendable.

With the latest changes on this branch it is not possible, since an upload is only scheduled when the last open file handle is closed and with https://github.com/cryptomator/fuse-cloud-access-adapter/commit/f07710b149583a65a450781d9265ba9ada71cc4f during this period it is not possible to open a new handle.

File: src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java (hunk 1 above)
Comment (infeo) [reply src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java:56]: Ok, `UUID:fromString` requieres a special format, forget that idea for now.

To be completely correct, i'll use a unique UUID.

Comment (infeo) [reply src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java]: Done with 1931539

File: src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java (hunk 2 above)
Comment (infeo) [reply src/main/java/org/cryptomator/fusecloudaccess/OpenFileUploader.java]: see 1931539

**Review by overheadhunter — APPROVED** [review]

**Pull request merged**